demo/midgard-sdk/src/endpoints/user-events/deposit/fetch.ts

```diff
@@ -0,0 +1,33 @@
+import { LucidEvolution, PolicyId, UTxO } from "@lucid-evolution/lucid";
+import { Deposit } from "@/tx-builder/index.js";
+import { Effect } from "effect";
+import { utxosAtByNFTPolicyId } from "@/utils/common.js";
+import { utxosToDepositUTxOs } from "@/utils/deposit.js";
+
+const isUTxOTimeValid = (
+  depositUTxO: Deposit.DepositUTxO,
+): Effect.Effect<Boolean, Error> =>
+  Effect.gen(function* () {
+    const { data: depositData } = depositUTxO.datum;
+    const currentTime = BigInt(Date.now());
+    return (
+      depositData.startTime <= currentTime && currentTime <= depositData.endTime
+    );
+  });
+
+export const fetchUnsortedDepositUTxOsProgram = (
```

Comment:
Deposit UTxOs are all isolated from one another, unlike the state queue's, which are nodes of a linked list.
```suggestion
export const fetchDepositUTxOsProgram = (
```